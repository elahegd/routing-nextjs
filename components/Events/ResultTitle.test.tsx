import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";

import ResultTitle from "./ResultTitle";

describe("Render and check props", () => {
  it("If button has link and children props", () => {
    const mockDate = new Date("2022-01-01");
    const { asFragment } = render(<ResultTitle date={mockDate} />);

    expect(asFragment()).toMatchSnapshot();
    const heading = screen.getByRole("heading");
    expect(heading.textContent).toBe("Events in January 2022");
  });

  it("Click on 'Show all events' link should redirect to /events page", () => {
    const mockDate = new Date("2022-01-01");
    render(<ResultTitle date={mockDate} />);

    const allEventsLink = screen.getByText("Show all events");
    fireEvent.click(allEventsLink);
    expect(allEventsLink.getAttribute("href")).toBe("/events");
  });
});
