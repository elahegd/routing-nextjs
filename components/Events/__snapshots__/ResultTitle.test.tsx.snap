// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render and check props If button has link and children props 1`] = `
<DocumentFragment>
  <section
    class="title"
  >
    <h1>
      Events in January 2022
    </h1>
    <a
      class="btn"
      href="/events"
    >
      Show all events
    </a>
  </section>
</DocumentFragment>
`;
